packages/gatsby/src/internal-plugins/query-runner/pages-writer.js

```diff
@@ -72,15 +72,17 @@ const writePages = async () => {
   components = _.uniqBy(components, c => c.componentChunkName)
 
   // Create file with sync requires of components/json files.
-  let syncRequires = `// prefer default export if available
+  let syncRequires = `const { hot } = require("react-hot-loader/root")
```

Comment:
All of this looks great--and I was able to validate that this does fix a number of issues with hot reloading + hooks, namely:

- Hot reloading of page components
- Hot reloading does not show an error on launch
- Hot reloading (still) works with class components, markdown files, etc.

However -- I don't think this is the best place to implement this hot reloading functionality. react-hot-loader is basically a noop in non-development mode (e.g. NODE_ENV=development), but I don't love the idea of putting this in front of every page in every environment.

@pieh do you have a better idea as to where this hot reloading functionality should be implemented?

Reply:
I agree, I originally had it in `app.js` (https://github.com/gatsbyjs/gatsby/pull/10455/commits/f0883f1385171b180fa11320cf95715f506f9c62) which also worked fine, but I moved it into the generated `sync-requires.js` as per https://github.com/gatsbyjs/gatsby/pull/10455#discussion_r241576619. 

Aren't the `syncRequires` only loaded in development mode anyways and replaced by `asyncRequires` in prod?

Reply:
@jgierer12 good catch! Yup!

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/production-app.js#L15

Reply:
> Aren't the `syncRequires` only loaded in development mode anyways and replaced by `asyncRequires` in prod?

Yes, they are currently - but we will be looking to split dev bundles similar to what we do for production - which should result in faster recompilations and also better experience for hosted/not-local preview servers. Not yet sure how would that affect hot reloads.